Name: Set Avatar in Messaging
Version: 2.0.1-1
Author: Jason Robitaille, dBsooner
Description: This adds a menu option to dynamically set avatar whenever you want. Requires FileMgr service.  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -252,6 +252,8 @@
 	},
 	
 	setup: function() {
+		var cookie = new Mojo.Model.Cookie("myAvatarPath");
+		this.myAvatar = cookie.get() || "images/list-view-default-avatar.png";
 		Mojo.Log.info("++++++ChatviewAssistant::setup");
 		this.controller.get('footer').hide();
 		this.checkThreadLocked();
@@ -2088,7 +2090,7 @@
 			if (isOutgoingMsg) {
 				tmpClass.push("my-chat");
 				// Disabling MY avatar because we are not doing this for 1.0  2008-11-05
-				//msg.avatar = this.myAvatar || 'images/my-avatar.png';
+				msg.avatar = this.myAvatar;
 			} else if (isTransientMsg || ChatFlags.isSystem(msg)) {
 				// isTransient is set on messages that we added to the chat when the account
 				// on the current transport goes online/offline
@@ -2097,7 +2099,7 @@
 				tmpClass.push("their-chat");
 				msg.avatar = this.avatarHash.get(msg);
 			}
-//			msg.imAvatarOverlay = "images/list-view-avatar-frame.png";
+			msg.imAvatarOverlay = "images/list-view-avatar-frame.png";
 
 			if(needsDeliveryReport && !hasDeliveryReport) {
 				// Waiting for a delivery report
--- .orig/usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
+++ /usr/palm/applications/com.palm.app.messaging/app/controllers/listview-assistant.js
@@ -256,6 +256,24 @@
 	},
 	
 	setup: function() {
+		var cookie = new Mojo.Model.Cookie("myAvatarPath");
+		this.myAvatar = cookie.get();
+		if(this.myAvatar) {
+			this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+				method: "exists",
+				parameters: {file: this.myAvatar},
+				onSuccess: function(response) {
+					if(!response.exists) {
+						cookie.remove();
+					}
+				}.bind(this)
+			});
+		}
+		this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+			method: "createDir",
+			parameters: {path: "/var/luna/files/"}
+		});
+		
 		this.appMenuAttrs = {
 			omitDefaultItems: true
 		};
@@ -738,6 +756,35 @@
 		}
 	},
 	
+	deleteAvatar: function() {
+		if(this.myAvatar) {
+			this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+				method: "delete",
+				parameters: {
+					file: this.myAvatar,
+				}
+			});
+		}
+	},
+	
+	copyAvatar: function(filePath) {
+		this.timestamp = new Date().getTime();
+		this.avatarExt = filePath.substring(filePath.lastIndexOf("."));
+		this.controller.serviceRequest("palm://ca.canucksoftware.filemgr", {
+			method: "copy",
+			parameters: {
+				from: filePath,
+				to: "/var/luna/files/_my-avatar-" + this.timestamp + this.avatarExt,
+			},
+			onSuccess: function() {
+				this.myAvatar = "/var/luna/files/_my-avatar-" + this.timestamp
+						+ this.avatarExt;
+				var cookie = new Mojo.Model.Cookie("myAvatarPath");
+				cookie.put(this.myAvatar);
+			}.bind(this)
+		});
+	},
+	
 	handleCommand: function(event) {
 		// handle menu button command events
 		if (event.type === Mojo.Event.command) {
@@ -741,6 +788,22 @@
 	handleCommand: function(event) {
 		// handle menu button command events
 		if (event.type === Mojo.Event.command) {
+			if (event.command === 'set-avatar') {
+				var self = this;
+				var params = {
+					extensions: ["bmp", "jpg", "jpeg", "png", "gif"],
+					kinds: ["image", "file"],
+					defaultKind: "image",
+					actionName: "Set Avatar",
+					onSelect: function(file){
+						self.deleteAvatar();
+						self.copyAvatar(file.fullPath);
+					}
+				};
+				Mojo.FilePicker.pickFile(params, this.controller.stageController);
+				event.stop();
+				return;
+			}
 			// determine if we have a target function for this command
 			var menu = $H(this.COMMAND_MENU);
 			menu.keys().each(function(key) {
@@ -775,6 +838,7 @@
 			enabled: true
 		};
 		var items = [Mojo.Menu.editItem];
+		items.push({label: "Set Avatar", command: "set-avatar"});
 		if (this.toEnableAddBuddyMenuItem === true) {
 			var appMenuAddBuddyItem = {
 				label: $L('Add Buddy'),
--- .orig/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ /usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -68,7 +68,7 @@
 	position: absolute;
 	top: 50%;
 	right: 0;
-	margin-top: -30px;
+	margin-top: -32px;
 }
 
 .login-animation-available,
@@ -386,7 +386,7 @@
 		height: 48px;
 		width: 48px;
 		border: 0;
-}
+}*/
 
 #myAvatar {
 		height: 34px;
@@ -397,7 +397,7 @@
 		background-color: #999;
 		border: 1px solid #000;
 }
-*/
+
 #listviewToggle {
 		position: fixed;
 		padding-left: 6px;
@@ -605,6 +605,7 @@
 
 .chat-entry.my-chat .chat-balloon {
 		-webkit-border-image: url(../images/chatview/chat-balloon-right.png) 10 10 10 10 stretch stretch;
+		min-height:33px;
 }
 
 .chat-entry.my-chat.error .chat-balloon, 
@@ -684,6 +685,10 @@
 		word-wrap: break-word !important;
 }
 
+.my-chat .chat-balloon-wrapper {
+		padding-right: 30px;
+}
+
 .chat-entry .message-body {
 		display: -webkit-box;
 		-webkit-box-orient: vertical;
@@ -693,13 +698,6 @@
 		min-height: 45px;
 }
 
-/* Hide user avatar */
-
-.my-chat .avatar-set,
-.my-chat .avatar-container {
-		display: none;
-}
-
 .message.status-success {
 		color: #000;
 }
\ No newline at end of file
